packages/dtslint-runner/src/check-parse-results.ts

```diff
@@ -1,217 +1,31 @@
-  TypingsData,
-  AllPackages,
-  formatTypingVersion,
-  getDefinitelyTyped,
-} from "@definitelytyped/definitions-parser";
-import { mapDefined, FS, logger, writeLog, Logger, ProgressBar, cacheDir, max, min } from "@definitelytyped/utils";
-import * as pacote from "pacote";
+import { AllPackages, getDefinitelyTyped } from "@definitelytyped/definitions-parser";
+import { assertDefined } from "@definitelytyped/utils";
 import * as semver from "semver";
-
 if (require.main === module) {
   const options = { definitelyTypedPath: undefined, progress: false, parseInParallel: false };
   getDefinitelyTyped(options, console).then((dt) => {
-    checkParseResults(/*includeNpmChecks*/ false, dt);
+    AllPackages.read(dt).then(checkParseResults);
   });
 }
 
-export async function checkParseResults(includeNpmChecks: false, dt: FS): Promise<void>;
-export async function checkParseResults(
-  includeNpmChecks: true,
-  dt: FS,
-  options: ParseDefinitionsOptions
-): Promise<void>;
-export async function checkParseResults(
-  includeNpmChecks: boolean,
-  dt: FS,
-  options?: ParseDefinitionsOptions
-): Promise<void> {
-  const allPackages = await AllPackages.read(dt);
-  const [log, logResult] = logger();
-
-  checkTypeScriptVersions(allPackages);
-
-  checkPathMappings(allPackages);
-
-  const dependedOn = new Set<string>();
-  const packages = allPackages.allPackages();
-  for (const pkg of packages) {
-    if (pkg instanceof TypingsData) {
-      for (const dep of Object.keys(pkg.dependencies)) {
-        dependedOn.add(dep);
-      }
-      for (const dep of pkg.testDependencies) {
-        dependedOn.add(dep);
-      }
-    }
-  }
-
-  if (includeNpmChecks) {
-    const allTypings = allPackages.allTypings();
-    const progress = options!.progress && new ProgressBar({ name: "Checking for typed packages..." });
-    let i = 0;
-    await Promise.all(
-      allTypings.map((pkg) =>
-        checkNpm(pkg, log, dependedOn).then(() => {
-          if (progress) progress.update(++i / allTypings.length, pkg.desc);
-        })
-      )
-    );
-    if (progress) progress.done();
-  }
-
-  await writeLog("conflicts.md", logResult());
-}
-
-function checkTypeScriptVersions(allPackages: AllPackages): void {
+export function checkParseResults(allPackages: AllPackages): string[] {
+  const errors = [];
   for (const pkg of allPackages.allTypings()) {
     for (const dep of allPackages.allDependencyTypings(pkg)) {
-      if (dep.minTypeScriptVersion > pkg.minTypeScriptVersion) {
-        throw new Error(`${pkg.desc} depends on ${dep.desc} but has a lower required TypeScript version.`);
+      // check raw version because parsed version doesn't currently retain range information
+      const version = assertDefined(new Map(pkg.allPackageJsonDependencies()).get(dep.name));
+      if (semver.parse(version)) {
+        errors.push(
+          `${pkg.desc}'s dependency on ${dep.desc}@${version} must use "*" or a version range, not a specific version.`
```

Comment:
Will this work for packages that were removed from DT but are being pinned back? I assume so given that weird express-brute-mongo thing works.

Reply:
This rule is in here because I observed that `"*"` is the only thing used and I figured I want to keep it that way until there is a concrete reason to allow other things. But it was "empirically derived" meaning that I didn't stop to consider *what* other things might eventually be allowed. It might still be nothing, with overrides specified via pnpm overrides.

Reply:
Well, CI should tell us if we do things wrong at least.